test/new-e2e/tests/npm/ec2_1host_test.go

```diff
@@ -97,176 +86,106 @@ func TestEC2VMSuite(t *testing.T) {
 	testURL := "http://" + v.Env().HTTPBinHost.Address + "/"
+	vt.Run("host_requests", func(t *testing.T) {
+		v.BaseSuite.SetT(t)
+		v.beforeTest("TestEC2VMSuite", t.Name()) // workaround as suite doesn't call BeforeTest before each sub tests
 
-	// generate a connection
-	v.Env().RemoteHost.MustExecute("curl " + testURL)
-
-	targetHostnameNetID := ""
-	// looking for 1 host to send CollectorConnections payload to the fakeintake
-	v.EventuallyWithT(func(c *assert.CollectT) {
-
-		hostnameNetID, err := v.Env().FakeIntake.Client().GetConnectionsNames()
-		assert.NoError(c, err, "GetConnectionsNames() errors")
-		if !assert.NotEmpty(c, hostnameNetID, "no connections yet") {
-			return
-		}
-		targetHostnameNetID = hostnameNetID[0]
+		// generate a connection
+		v.Env().RemoteHost.MustExecute("curl " + testURL)
 
-		t.Logf("hostname+networkID %v seen connections", hostnameNetID)
-	}, 60*time.Second, time.Second, "no connections received")
+		test1HostFakeIntakeNPM(&v.BaseSuite, v.Env().FakeIntake)
+	})
+	vt.Run("docker_requests", func(t *testing.T) {
+		v.BaseSuite.SetT(t)
+		v.beforeTest("TestEC2VMSuite", t.Name()) // workaround as suite doesn't call BeforeTest before each sub tests
 
-	// looking for 3 payloads and check if the last 2 have a span of 30s +/- 500ms
-	v.EventuallyWithT(func(c *assert.CollectT) {
-		cnx, err := v.Env().FakeIntake.Client().GetConnections()
-		assert.NoError(t, err)
+		// generate a connection
+		v.Env().RemoteHost.MustExecute("docker run curlimages/curl curl " + testURL)
 
-		if !assert.Greater(c, len(cnx.GetPayloadsByName(targetHostnameNetID)), 2, "not enough payloads") {
-			return
-		}
-		var payloadsTimestamps []time.Time
-		for _, cc := range cnx.GetPayloadsByName(targetHostnameNetID) {
-			payloadsTimestamps = append(payloadsTimestamps, cc.GetCollectedTime())
-		}
-		dt := payloadsTimestamps[2].Sub(payloadsTimestamps[1]).Seconds()
-		t.Logf("hostname+networkID %v diff time %f seconds", targetHostnameNetID, dt)
-
-		// we want the test fail now, not retrying on the next payloads
-		assert.Greater(t, 0.5, math.Abs(dt-30), "delta between collection is higher than 500ms")
-	}, 90*time.Second, time.Second, "not enough connections received")
+		test1HostFakeIntakeNPM(&v.BaseSuite, v.Env().FakeIntake)
+	})
 }
 
 // TestFakeIntakeNPM_600cnx_bucket Validate the agent can communicate with the (fake) backend and send connections
 // every 30 seconds with a maximum of 600 connections per payloads, if more another payload will follow.
 //   - looking for 1 host to send CollectorConnections payload to the fakeintake
 //   - looking for n payloads and check if the last 2 have a maximum span of 100ms
 func (v *ec2VMSuite) TestFakeIntakeNPM_600cnx_bucket() {
-	t := v.T()
+	vt := v.T()
 	testURL := "http://" + v.Env().HTTPBinHost.Address + "/"
+	vt.Run("host_requests", func(t *testing.T) {
+		v.BaseSuite.SetT(t)
+		v.beforeTest("TestEC2VMSuite", t.Name()) // workaround as suite doesn't call BeforeTest before each sub tests
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
Rather than `beforetest`, I would expect to call directly the fakeintake flush: does it work out of devmode ?

Reply:
If these tests do not need to run in sequence, you can use suite tests instead

Reply:
beforeTest() is here because subtests are not calling BeforeTest() so all tests would have fakeintake flushed by default (but not in devmode)
Order is not important but I tried to limit the copy/paste and test will report name `TestSuite/Test1/host_requests`